interpreter/src/ser.rs

```diff
@@ -561,31 +561,6 @@ impl ser::Serializer for KeySerializer {
     }
 }
 
-impl Serialize for Value {
```

Comment:
I'm pretty sure it was a poorly thought out solution for allowing the add_variable method to support both serializable types and values -- by making values serializable. The more I think about it, the more I'm comfortable with my changes here, I think they accomplish the same objective.

Reply:
Cool! Live and learn (and rinse & repeat! 😉)

Reply:
Hi, the removal of this broke https://github.com/clarkmcc/cel-rust/pull/40.  What I've been doing is calling:

```rust
let result: serde_json::Value = serde_json::to_value(value).map_err(|err| anyhow!("Unable to serialize JSON value: {}", err))?;
```

Where `value` is a `cel_interpreter::Value`. Is there a better/easier way I can convert a CEL `Value` into a `serde_json::Value`?

Reply:
So there is one thing I'm wondering here... _I think_ it relates [this issue](https://github.com/clarkmcc/cel-rust/issues/73), ... and I could be wrong as I haven't looked how the golang reference implementation goes about it. But shouldn't these things be considered "together" and have a "symmetric" solution to get `Value`s in & out of "the CEL space" into others? Mostly thinking protobuf and JSON, as these are called out explicitly in the language spec? Feel free to ignore, as I know I don't have a handle on the problem space quite yet, but I know I would like to dig into it (as my use case is CEL & Protobuf).

Reply:
@alexsnaps I think you're on the right track -- when you want get a strongly typed output, I don't believe the correct solution here is to serialize the CEL output to JSON, then deserialize from JSON to a struct. I think we want some mechanism to turn `Value` into structs and then you can do whatever you want with them. I _don't know_ if that means introducing interpreter-supported structs like #73 as a solution, or if we just need to provide a custom `Deserialize` implementation for `Value` which would be format-agnostic.

Looking at cel-go this is what you get back as the result of an expression. You wouldn't serialize this directly to JSON either. You'd switch on type, or you'd convert directly to a Go struct that was registered to the interpreter.
https://github.com/google/cel-go/blob/37d5b961f5da8ca2a21ccb565af63faaf84ed2fa/common/types/ref/reference.go#L37-L63

Reply:
@jplock if you haven't already, you might pin your dependency to `0.8.0` to work around this change until we can nail down the API for this sort of thing going forward. Thanks for bringing up the use-case!

Reply:
@jplock would you mind giving [this branch](https://github.com/clarkmcc/cel-rust/pull/77) a try?

Reply:
I'll throw out there that have a use case that's broken by this change as well: I'm using serde to avoid thousands of lines of binding code for conversion of many structs to CEL (I haven't quite wrapped my head around writing custom marcos yet), but there are a few fields that accept arbitrary input (as long as it can be represented in both TOML and CEL). I was using `cel_interpreter::Value` as the type for those fields, but after updating these no longer work with the `Serialize` derive.

For now I'll just create a wrapper type and copy this removed implementation to my project, but fwiw the ability to nest `Value` in a type and use the `Serialize` derive was useful to me, so much so that I wrote a PR to fix almost the exact issue this PR did except when using serde